docs/050-breaking-changes.rst

```diff
@@ -0,0 +1,379 @@
+
+* Functions without implementation cannot use modifiers anymore.
+
+* Function types with named return values are now disallowed.
+
+* Single statement variable declarations inside if/while/for bodies that are
+  not blocks are now disallowed.
+
+Example
+=======
+
+The following example shows a contract and its updated version for Solidity
+v0.5.0 with some of the changes listed in this section.
+
+Old version:
+
+::
+
+   pragma solidity ^0.4.25;
+
+   contract OtherContract {
+      uint x;
+      function f(uint y) external {
+         x = y;
+      }
+      function() payable external {}
+   }
+
+   contract Old {
+      OtherContract other;
+      uint myNumber;
+
+      function someInteger() internal returns (uint) { return 2; }
+
+      // Function visibility not provided, not an error.
+      // Function mutability not provided, not an error.
+      function f(uint x) {
+         // Var is fine in this version.
+         var z = someInteger();
+         x += z;
+         // Throw is fine in this version.
+         if (x > 100)
+            throw;
+         y = -3 >> 1;
+         // y = -1 (wrong, should be -2)
+         do {
+            x += 1;
+            if (x > 10) continue;
+            // 'Continue' causes an infinite loop.
+         } while (x < 11);
+         // Call returns only a Bool.
+         bool success = other.call("f");
+         if (!success)
+            revert();
+         else {
+            // Local variables could be declared after their use.
+            int y;
+         }
+      }
+
+      // No need for an explicit data location for 'arr'
+      function g(uint[] arr, bytes8 x, OtherContract otherContract) public {
+         otherContract.transfer(1 ether);
+
+         // Since uint32 (4 bytes) is smaller than bytes8 (8 bytes),
+         // the first 4 bytes of x will be lost. This is dangerous
+         // because bytesX are right padded.
+         uint32 y = uint32(x);
+         myNumber += y + msg.value;
+      }
+   }
+
+New version:
+
+::
+
+   pragma solidity >0.4.25;
+
+   contract OtherContract {
+      uint x;
+      function f(uint y) external {
+         x = y;
+      }
+      function() payable external {}
+   }
+
+   contract New {
+      OtherContract other;
+      uint myNumber;
+
+      function someInteger() internal returns (uint) { return 2; }
+
+      // Function visibility must be specified.
+      // Function mutability must be specified.
+      function f(uint x) public returns (bytes memory) {
+         // The type must now be explicitly given.
+         uint z = someInteger();
+         x += z;
+         // Throw is now disallowed.
+         require(x > 100);
+         int y = -3 >> 1;
+         // y = -2 (correct)
+         do {
+            x += 1;
+            if (x > 10) continue;
+            // 'Continue' jumps to the condition below.
+         } while (x < 11);
+
+         // Call returns (bool, bytes).
+         // Data location must be specified.
```

Comment:
Have to first convert `other` to `address` to get access to the `call` function.